client/landing/gutenboarding/onboarding-block/style-preview/disableNavigation.ts

```diff
@@ -0,0 +1,43 @@
+/**
+ * Changes href attributes' values to "#!" and removes meta-redirect tags.
+ *
+ * @param contentDocument the document you want to process
+ */
+export function disarmAnchorsAndMetaRefreshes( contentDocument: HTMLDocument ) {
+	// dis-arm anchors
+	contentDocument.body
+		?.querySelectorAll( 'a[href]:not([href="#!"])' )
+		.forEach( ( anchor: Element ) => {
+			// we don't want to remove href attribute because styling can change without it
+			// using only # will jump to top
+			anchor.setAttribute( 'href', '#!' );
+		} );
+
+	// remove meta redirects
+	contentDocument.head
+		?.querySelectorAll( 'meta[http-equiv="refresh"]' )
+		.forEach( ( element: Element ) => {
+			element.parentNode?.removeChild( element ); // element.remove() doesn't work in IE
+		} );
+}
+
+/**
+ * Handles all submit events in hte iframe and prevents their default behavior.
+ * Also it handles `load` event for the iframe and calls {@link disarmAnchorsAndMetaRefreshes} on the iframe document when it fires
+ *
+ * @param iframe The iframe you want to process
+ */
+export function disableNavigation( iframe: HTMLIFrameElement ) {
+	const { contentDocument, contentWindow } = iframe;
+
+	if ( contentDocument && contentWindow ) {
+		// submit event bubbles up, we only need to catch it on the window level
+		contentWindow.addEventListener( 'submit', ( event: Event ) => {
+			event.preventDefault();
+			return false;
+		} );
+
+		// run again after the iframe loads
```

Comment:
I'm not familiar with the flow, but curious why you need to call this again - or would it be enough to just call it once here (when iframe loads)?

Reply:
#### I think of it like this:

1. **Before `onload`**: process the HTML and remove server-rendered links. This is neccessary because the user can click before the iframe is fully loaded. If we only process after `onload` event, clicks will actually work for the whole load time, can be 5 seconds. 
2. **After `onload`**: If there a link that is constructed by JS, for example, a jQuery hamburger menu or a React widget or anything JS rendered, we'll need to remove it after `onload` because it won't exist in the initial HTML.

Reply:
You're right! But `window.location` is read only we cann't do anything about it.